packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+
+import {
+  default as ConfirmationModal,
+  Variant as ConfirmationModalVariant,
+} from '@leafygreen-ui/confirmation-modal';
+
+export { ConfirmationModalVariant };
+
+type ConfirmationProperties = {
+  title: string;
+  description: string;
+  buttonText?: string;
+  variant?: ConfirmationModalVariant;
+};
+type ConfirmationCallback = (value: boolean) => void;
+
+interface ConfirmationModalContextData {
+  showConfirmation: (
+    props: ConfirmationProperties,
+    callback: ConfirmationCallback
+  ) => void;
+  isMounted: boolean;
+}
+
+const ConfirmationModalContext =
+  React.createContext<ConfirmationModalContextData>({
+    isMounted: false,
+  } as ConfirmationModalContextData);
+
+class ShowConfirmationEvent extends Event {
+  constructor(
+    public props: ConfirmationProperties,
+    public callback: ConfirmationCallback
+  ) {
+    super('show-confirmation');
+  }
+}
+class GlobalConfirmation extends EventTarget {
+  showConfirmation(props: ConfirmationProperties) {
+    return new Promise<boolean>((resolve) => {
+      const event = new ShowConfirmationEvent(props, resolve);
+      this.dispatchEvent(event);
+    });
+  }
+}
+const globalConfirmation = new GlobalConfirmation();
+export const showConfirmation =
+  globalConfirmation.showConfirmation.bind(globalConfirmation);
+
+type ConfirmationModalAreaProps = ConfirmationProperties & {
+  open: boolean;
+};
+
+export const ConfirmationModalArea: React.FC = ({ children }) => {
+  const hasParentContext = useContext(ConfirmationModalContext).isMounted;
+
+  const [confirmationProps, setConfirmationProps] = useState({
+    open: false,
+  } as ConfirmationModalAreaProps);
+  const callbackRef = useRef<(value: boolean) => void>();
+
+  useEffect(() => {
+    return () => {
+      callbackRef.current?.(false);
+    };
+  }, []);
+
+  // Event listener to use confirmation modal outside of react
+  useEffect(() => {
+    const listener = ({ props, callback }: ShowConfirmationEvent) => {
+      showConfirmation(props, callback);
+    };
+    globalConfirmation.addEventListener(
+      'show-confirmation',
+      listener as EventListener
+    );
+    return () => {
+      globalConfirmation.removeEventListener(
+        'show-confirmation',
+        listener as EventListener
+      );
+    };
+  }, []);
+
+  const showConfirmation = (
+    props: ConfirmationProperties,
+    callback: ConfirmationCallback
+  ) => {
+    if (confirmationProps.open) {
+      throw new Error('Confirmation modal is already open');
+    }
```

Comment:
[This issue](https://github.com/mongodb-js/compass/pull/4079#discussion_r1113042661) that I noted still exists, the global method and the one passed through context have different behavior, which is not great, either both should throw if modal is already open (which is probably a preferred behavior) or both should override current confirmation, we should also document what will happen